play.pokemonshowdown.com/src/panel-battle.tsx

```diff
@@ -526,15 +590,29 @@ class BattlePanel extends PSRoomPanel<BattleRoom> {
 	}
 	renderSwitchControls(request: BattleMoveRequest | BattleSwitchRequest, choices: BattleChoiceBuilder) {
 		const numActive = choices.requestLength();
-
+		const maybeTrapped = choices.currentMoveRequest()?.maybeTrapped;
 		const trapped = choices.currentMoveRequest()?.trapped;
 
-		return request.side.pokemon.map((serverPokemon, i) => {
-			const cantSwitch = trapped || i < numActive || choices.alreadySwitchingIn.includes(i + 1) || serverPokemon.fainted;
+		return (<div> {(trapped || maybeTrapped) &&
+			(maybeTrapped ?
+				<em class="movewarning">You <strong>might</strong> be trapped, so you won't be able to cancel a switch! <br /></em> :
+				<em class="movewarning">You <strong>are</strong> trapped and cannot switch! <br /></em>)}
```

Comment:
```suggestion
				<em class="movewarning">You're <strong>trapped</strong> and cannot switch!<br /></em>)}
```

This message does not necessarily appear after a `maybeTrapped` (for instance, Fire Spin will just directly trap you), so bolding "are" could be confusing here.